packages/lib/db-commons/index.cjs

```diff
@@ -137,6 +137,13 @@ const asyncIterableToBatchedAsyncGenerator = async function (
 		let null_columns = column_names.filter((column) => firstRow[column] == null);
 		while (null_columns.length > 0) {
 			const next = await iterator.next().then((x) => x.value);
+
+			// When the iterator is done, next will be { done: true, value: undefined }
+			// We want to break out of the loop when we reach the end of the iterator
+			if (!next) {
```

Comment:
This comment conflicts with your `if` statement. If `next` is `{ done: true, value: undefined }` when the iterator is done, shouldn't we check `if (next.done)` instead of `if (!next)`?

Reply:
As a note, the case where `{ done: true, value: ["some value"] }` is possible if a `return ["some value"]` is used (rather than just `return`)

Reply:
If you consider this example, one of the generator functions has a return value, the other doesn't. The one that has a return value does not have that value used the same way as the yielded values in `evalIter` when only testing on `x.done` - done is true, and a value is present

```javascript
function* iteratorWithoutReturnValue() {
  for (let i = 0; i < 10; i++) {
    // Yields 0 -> 9
    yield i;
  }
}

function* iteratorWithReturnValue() {
  for (let i = 0; i < 10; i++) {
    // Yields 0 -> 9
    yield i;
  }
  // Returns a string
  return -1
}

function evalIter(iter) {
  let x;
  while((x = iter.next()), !x.done) {
    console.log(x)
  }
  // If there is a value in x, then it will not be used correctly
  console.log("Loop done, x = ", x)  
}


// Will end with `Loop done, x =  { value: undefined, done: true }`
evalIter(iteratorWithoutReturnValue())

// Will end with `Loop done, x =  { value: -1, done: true }`
evalIter(iteratorWithReturnValue())
```

Reply:
@ItsMeBrianD You are right. Like i said, i tried to change as little code as possible since I'm not familiar with the codebase. But just sent a commit that applied your changes. Please let me know if that's enough.